django/api/views.py

```diff
@@ -1,27 +1,43 @@
-from rest_framework import generics
+from rest_framework import generics, mixins, status
 from rest_framework.permissions import IsAuthenticated
-from rest_framework import mixins
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
 from rest_framework.viewsets import GenericViewSet
 from rest_framework_gis.filters import InBBoxFilter
 from django.contrib.auth.models import User
 from api.models import Restaurant
-from api.serializers import UserSerializer, RestaurantSerializer
+from api.serializers import UserSerializer, RestaurantReadSerializer, CreateUserWithRestaurantSerializer
 
-
-class UserViewSet(mixins.RetrieveModelMixin, GenericViewSet):
+class UserList(generics.ListAPIView):
     permission_classes = (IsAuthenticated,)
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
+#Create User and User as Restaurant owner with no permissions
+class CreateUserView(generics.CreateAPIView):
+    model_class = User
+
     def get_object(self):
         return self.request.user
 
+    def get_serializer_class(self):
+        if self.request.data.__contains__("restaurant"):
+            return CreateUserWithRestaurantSerializer
+        return UserSerializer
+
+#Interact with User with permissions
+class UserViewSet(mixins.RetrieveModelMixin, mixins.UpdateModelMixin, GenericViewSet):
```

Comment:
Nice use of multiple inheritance here. I would probably put the `GenericViewSet` first in the list -- the order does matter for multiple inheritance although in this case I don't think it'll make a difference.